.bazelci/presubmit.yml

```diff
@@ -92,8 +92,10 @@ tasks:
     test_targets: *default_linux_targets
     build_flags:
       - "--compilation_mode=opt"
+      - "--@rules_rust//rust/settings:experimental_toolchain_generated_sysroot=True"
```

Comment:
Instead of updating the CI jobs, can you update a unit tests to transition this setting? Or add it to the `.bazelrc`, but in general I would like `presubmit.yaml` to be used as an absolute last resort.

Reply:
How will eventually roll out (e.g. remove the bool flag)?

Reply:
To me, the unit tests aren't really useful in this context because we're trying to verify whether `rustc` executes successfully. The unit tests (aka Bazel's analysis test) only verifies that the action has the correct argv (`--sysroot`) but doesn't really verify that `rustc` builds the source successfully.

Reply:
https://github.com/bazelbuild/rules_rust/blob/main/COMPATIBILITY.md#how-to-make-a-backwards-incompatible-change is the process we should probably follow here.



> IMO, the unit tests aren't really useful in this context because we're trying to verify whether `rustc` executes successfully given the new `--sysroot` flag pointing to a generated directory. The unit tests (aka Bazel's analysis tests) only verify the action has the correct argv (`--sysroot`) but don't verify that `rustc` builds the source successfully.

The analysistest has [config_settings](https://github.com/bazelbuild/bazel-skylib/blob/main/docs/unittest_doc.md#analysistest.make-config_settings) which would allow you to transition `target_under_test` to enable this specific feature. I think largely we'd just wanna make sure `--sysroot` appears in the Rustc invocation (which would probably still be a test we keep after the flag is deleted). We could cut a release while the flag is disabled and then quickly enable it so users who like to stick closer to HEAD can identify issues. What do you think about that?

Reply:
The analysis tests will ensure `--sysroot` exists in the action command (i.e. **analysis phase**). However, it does not ensure whether `rustc` uses the stdlibs and binaries to build end users's rust code successfully (i.e. **execution phase**).

Having it built successfully on CI makes sure that rust code in `rules_rust` all build successfully with `--sysroot` being set. So on my mind, the rollout phases will be the following:
1. Add a build setting per the instruction in https://github.com/bazelbuild/rules_rust/blob/main/COMPATIBILITY.md#how-to-make-a-backwards-incompatible-change
2. Update analysis tests to enable the build setting
3. Update CI to build all rules_rust's code with the build setting enabled. (Rollout to `rules_rust`'s code)
4. Remove the build setting. (Rollout to end users's code)

(3) ensures that isn't any regression in execution phase before rolling it out to end users.

Let me know what you think.

Reply:
PTAL https://github.com/bazelbuild/rules_rust/pull/2223/commits/d985ad88cb7c183e3e5789c7160fc94eaafd3f74